node/silkworm/db/mdbx.hpp

```diff
@@ -58,10 +58,24 @@ class RWTxn {
     mdbx::txn& operator*() { return external_txn_ ? *external_txn_ : managed_txn_; }
     mdbx::txn* operator->() { return external_txn_ ? external_txn_ : &managed_txn_; }
 
-    void commit() {
+    void commit(const bool renew = true) {
+        /*
+         * renew is required here due to RAII
+         * RWTxn txn(env);
+         * txn.commit();
+         * env.close();
```

Comment:
Why call `env.close()` explicitly? If we rely on `env`'s destructor, `env` will be closed after `tnx`'s destructor.

Reply:
This is a comment only. It explains the rationale behind the implementation of `renew`
Also : should we know the object is no longer needed after the last commit is convenient NOT to open a new transaction for it to be aborted when the object falls out of scope.
Besides it might be necessary to call `env.close()` explicitly for example:
- to measure fsync timings (not yet implemented)
- to trap exceptions (see silkworm.cpp)